be/src/olap/rowset/segment_v2/inverted_index_reader.h

```diff
@@ -64,7 +64,7 @@ enum class InvertedIndexReaderType {
     BKD = 2,
 };
 
-class InvertedIndexReader {
```

Comment:
If you enable share from this. Then I think you has to add macro ENABLE_FACTORY_CREATOR to disable explicit new .